src/web/src/hooks/use-telescope-contributor.ts

```diff
@@ -12,33 +13,59 @@ export type Contributor = {
 const telescopeGitHubContributorsUrl =
   'https://api.github.com/repos/Seneca-CDOT/telescope/contributors';
 
-const randomContributor = (contributors?: any[]) =>
-  contributors ? contributors[Math.floor(Math.random() * contributors.length)] : null;
+const prepareURL = (page?: number): string =>
+  `${telescopeGitHubContributorsUrl}?per_page=1&${page ? `page=${page}` : ''}`;
 
-const PER_PAGE: number = 50;
+const getPage = (links: string, rel: string): string | undefined => {
+  // Header link is the form  '<...?per_page=1&page="page">; rel="key">', last page is also the contributors count
+  const regex = new RegExp(`<[^?]+\\?per_page=1&[a-z]+=([\\d]+)>;[\\s]*rel="${rel}"`, 'g');
+  const arr = regex.exec(links);
+  return arr?.[1];
+};
+
+// We'll update github contributors count once every 30 days
+const maxAge = 30 * 24 * 60 * 60 * 1000;
 
 const useTelescopeContributor = () => {
-  const [contributor, setContributor] = useState<Contributor | null>(null);
+  const [maxContributors, setMaxContributors] = useLocalStorage<number>(
+    'github_contributors',
+    undefined,
+    {
+      raw: true,
+    }
+  );
+  const [fetchDate, setFetchDate] = useLocalStorage<number>('fetch_date', undefined, { raw: true });
+  const [random, setRandom] = useState<number>(-1);
 
-  const { data, error, size, setSize } = useSWRInfinite(
-    (index: number) => `${telescopeGitHubContributorsUrl}?page=${index + 1}&per_page=${PER_PAGE}`
+  const { data: lastPage, error: countError } = useSWR(
+    () =>
+      !maxContributors || !fetchDate || Date.now() - fetchDate >= maxAge ? prepareURL() : null,
```

Comment:
In the future, I would define a named function (eg., inline const) so that this code is more readable, and the name of the function documents what's happening here.  You have a lot of logic compressed in this, and to understand the code, you have to unpack it all.  A named function would eliminate the need for this.